/* -------------------------------------------------------------------------- */
/* Copyright 2002-2025, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/* -------------------------------------------------------------------------- */
syntax = "proto3";

import "shared.proto";

package one.acl;

option go_package = "github.com/OpenNebula/one/src/oca/go/src/goca/api/acl";

message AddRuleRequest
{
  string session_id = 1;
  string user       = 2;
  string resource   = 3;
  string rights     = 4;
  string zone       = 5;
}

message DelRuleRequest
{
  string session_id = 1;
  int32 oid         = 2;
}

message InfoRequest
{
  string session_id = 1;
}

service AclService
{
  rpc AddRule (one.acl.AddRuleRequest) returns (one.ResponseID);

  rpc DelRule (one.acl.DelRuleRequest) returns (one.ResponseID);

  rpc Info (one.acl.InfoRequest) returns (one.ResponseXML);
}
